### DISCUSSION TITLE
[FeatureRquest] NFC

### DISCUSSION THREAD

**Discussion by bademux** [joeycastillo/Sensor-Watch#52] [Ideas]
Hi,

I think it would be nice to have wireless communication with the world. As far as I understand NFC will be the only and the best choice here considering power usage and it's availability. I'm not sure why NFC is not considered in any of opensource watches.

Implementing NFC as sensor board was discussed here https://github.com/joeycastillo/Sensor-Watch/discussions/31

In my opinion NFC can greatly enhance TOTP feature.

**Comment by joeycastillo** [comment]
I would love to see it, I just lack the expertise in terms of knowing whether an on-board NFC antenna would work within the constraints we have. I did a little test board in Eagle and cut it out on paper; I get the sense that there's about 9.4 mm x 8.7 mm available for the antenna. The top layer of the flex PCB can be available for this purpose, and it would sit maybe a few millimeters below the screen. If anyone has a sense for whether this would be workable, I'd love to get an opinion. The board file is [OSO-MISC-22-004 WIP-NFC](https://github.com/joeycastillo/Sensor-Watch/blob/main/PCB/Sensor%20Boards/OSO-MISC-22-004%20WIP-NFC.brd).

![image](https://user-images.githubusercontent.com/1765152/152691998-f7b4fbfc-e82e-4a0e-a4bf-2af68cc765bc.png)

**Comment by bademux** [comment]
the same here, I'm not able to engineer good (any? :) ) NFC antenna.
waiting for someone who can help us and tell can we design passive antenna or are we able to use something like [this](https://www.mouser.pl/new/taoglas/taoglas-naa5point5/) on top of MCU.
Another idea will be to use custom circular antenna embedded into custom watch glass, but it is still my fantasy...

**Comment by ccoenen** [comment]
The Solo v2 ([software](https://github.com/solokeys/solo2)/[hardware](https://github.com/solokeys/solo2-hw)) has NFC and if I recall correctly it was [not easy getting this right](https://www.kickstarter.com/projects/conorpatrick/solo-v2-safety-net-against-phishing/posts/3163328). So I think their antenna size could be considered as a bar you'd need to clear. Sadly, the antenna is not part of the hardware repository :-(

**Comment by nrg-fv** [comment]
This is probably the answer to adding NFC support, at least antenna design is available. I'm not sure if any additional hardware would be required (probably yes) and if this all can be somehow connected to the board. I hope someone with better knowledge can explain.

https://n-o-d-e.net/datarunner.html

**Comment by bademux** [comment]
looks like NFC antenna behind the LCD will not work (https://youtu.be/aGn2_zP8GI8?t=66).

**Reply by 814d3** [reply]
The showed and working solution seems to be practicable. The thin antenna can be placed "somewhere" but the wiring could be tricky - or am I wrong?

**Reply by ccoenen** [reply]
also a larger area would likely work even behind the display. The sticker shown in the video is _tiny_ and antenna size does matter.

**Reply by bademux** [reply]
well, as far as I understand there is no place on the board for antenna (?).
And for compatibility LCD is the only place: watch body can be metallic.

**Comment by trash-anger** [comment]
Imagine being able to flash OTA from NFC! 😄 (no more usb!)

**Comment by ottomorac** [comment]
This idea would be cool. I love the idea of using the OTPs as currently implemented, but having a FIDO2 style solution with NFC would be amazing!  Essentially think of a Yubikey with NFC inside the SensorWatch. 

![image](https://github.com/user-attachments/assets/dc25d232-a009-4a27-93bf-fa5edc35b9a4)

**Comment by eerii** [comment]
Something awesome would be to have a Fidesmo Pay chip (https://fidesmo.com/consumer/fidesmo-pay/), having tap to pay on the watch. I'm pretty sure this would be more difficult than just adding an NFC chip, but it may be worth an investigation

**Reply by buster** [reply]
This would. Be awesome!
I know people using pagopace rings to pay and it's awesome. Would be even more awesome to pay with a Casio watch

**Comment by truedat101** [comment]
I have access to a team that is building tags and antenna stuff, and I am working on a no code thing for this company to deploy tags.  Maybe the NFC tag in a strap.  You just need a little dipole thingy to hook into the MCU.  Honestly, the watch footprint is super easy (I gave my NFC watch-strap samples to a colleague here in Japan otherwise I would send a picture), but the problem that others pointed out is the antenna would block the screen, and can't really sit behind the screen.    See the SEEED XIAO, there's a way to wire in an NFC antenna to be able to create a reader.

What is crazy funny in Japan  is people here don't realize they have NFC everywhere, and have had it for years.  When I showed a no-code tag solution, it was sort of shock and amazement.  I'm like, you guys have NFC everywhere.... It's just that everything is a closed system.

I can pass the photo along to the CEO of the client I work for to get comments about the antenna design.  He's seen most things that have been proposed over the years NFC and RFID and will have some ideas.  

Some design notes (I'm not a hardware RF engineer so ....)

- quality of your scan area is affected by the antenna size and other factors, but size and shape are a big deal.  The typical casio watch about matches the normal size of an antenna.  My observation working with different sized tags.
- Antennas that are sitting on an electrically conductive surface or on metal need a special design.  This is why an NFC tag on your macbook doesn't work... you need a special tag that has the insulation or whatever in it's design.  
- Space between the reader and tag has to be small .   A big old case gets in the way .
- There is a spec for long range tags (long range being like feet and not fractions of inches), but I don't know how widely this is out there, and won't work with phones.
- I've sat and watched people try to pay for their subway rides with their iwatches ... it usually works, but sometimes it doesn't probably due to special cases or band accessories.  Normally it just works.  When it doesn't, they have to give up and pull out cash.

**Comment by truedat101** [comment]
Also ... just a design comment.  An NFC tag that can also act as a reader is doubly interesting.  This is what your iphone or android is.  It can be a tag and a reader and a writer.  Also, the other meta thing about NFC is that you can write dynamic data into a tag if it is connected to power and some intelligence.  So it is a way to pass along diagnostic data, share pairing info, or my favorite old school thing, share a vcard.  The payment stuff mentioned is cool, but requires some licensing I think.  Tag ICs have their own IP that can be unlocked by the company that makes them.  That's a software thing.  For example, STM has the ST25 line, and there are special security features available, but some stuff is only available to licensees under NDA.  So if you are building and want to use those features, it is probably some royalty arrangement.
